index.bs

```diff
@@ -200,11 +200,50 @@ th {
 
 th, td {
   border-bottom: 1px solid black;
-  border-collapse: collapse;
+  border-collapse: collapse; /* BUG: This property only applies to TABLE. */
   padding-left: 5px;
   padding-right: 5px;
 }
 
+/*
+ * Grid table format.
+ */
+table.grid {
+  border-collapse: collapse;
+  table-layout: fixed;
+  width: 100%;
+}
+
+table.grid td, table.grid th {
+  border: 1px solid black;
+}
+
+table.grid th {
+  text-align: center;
+}
+
+/* For a table header cell that is split diagonally */
+th.split {
+  background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' preserveAspectRatio='none' viewBox='0 0 1 1'><line x1='0' y1='0' x2='1' y2='1' stroke='black' vector-effect='non-scaling-stroke'/></svg>");
```

Comment:
Is all the SVG just for the sake of the single pixel thick stroke? I wonder if there's a pure CSS way to achieve it.

Reply:
Yes, just for the line. Simpler approaches welcome!

Reply:
So a helpful bot told me that we could simply use the following CSS...

```
  vector-effect: non-scaling-stroke;
  stroke-width: 1; /* Set your desired stroke width */
```

...but then LLM's fabricate -_-.

Is it sufficient to just use a *thin* border style? For very high DPI monitors, maybe we *don't* want to enforce a single pixel always.

```
  border-style: solid;
  border-width: thin;
```

Reply:
What I was using was inspired by https://www.peterkrautzberger.org/0213/ Breaking down what `th.split` is doing in this PR:

```css
th.split {                                                           
  background-image: url("data:image/svg+xml;utf8,...");  /* 👈 image of a diagonal line */
  background-repeat: no-repeat; /* 👈 don't tile, and start at top left */
  background-size: 100% 100%; /* 👈 scale to cover the entire table cell */
  position: relative; /* 👈 make positioned, so child labels can be absolutely positioned within it */
}           
```

And breaking down the SVG:

```xml
<svg xmlns='http://www.w3.org/2000/svg' version='1.1'  👈 boilerplate
  preserveAspectRatio='none'   👈 allow stretching to cover entire table cell
  viewBox='0 0 1 1'> 👈 defines our coordinate space; don't need much
    <line 👈 render a single line
       x1='0' y1='0' x2='1' y2='1' 👈 from top-left to bottom-right
       stroke='black'  👈 default wouldn't be visible
       vector-effect='non-scaling-stroke' 👈 we don't want the stroke to be wider or thinner when the viewbox is stretched
    />
</svg>"
```

So... this seems pretty minimal to me, and I'm not sure how the suggested CSS helps. The stroke width's default of 1px can be overridden, e.g. `stroke-width='5px'` but that doesn't eliminate the need for the any of the above - without the `vector-effect` the image being stretched to cover the table cell's arbitrary size means the stroke would scale strangely.

Reply:
Oops, I didn't realize that was a *diagonal* line, misreading it as vertical. (the full breakdown was not necessary, but thanks) - I could have just copied that snippet into my little [SVG Viewer](https://github.com/fdwr/LunaSvgSampleTest) to see what it produced  :b.